types/abbrev/index.d.ts

```diff
@@ -5,8 +5,8 @@
 
 export = abbrev;
 
-declare function abbrev(words: string[]): { [abbreviation: string]: string };
-declare function abbrev(...words: string[]): { [abbreviation: string]: string };
+declare function abbrev(words: Array<string | { toString(): string }>): { [abbreviation: string]: string };
+declare function abbrev(...words: Array<string | { toString(): string }>): { [abbreviation: string]: string };
```

Comment:
Please extract the parameter type into a separate named type (`Abbreviable` in the example). Please also use `ReadonlyArray` instead of array for input parameters that are not changed by the `abbrev` function:

```ts
declare function abbrev(words: ReadonlyArray<abbrev.Abbreviable>): { [abbreviation: string]: string };
declare function abbrev(...words: ReadonlyArray<abbrev.Abbreviable>): { [abbreviation: string]: string };

declare namespace abbrev {
    function monkeyPatch(): void;

    type Abbreviable = string | { toString(): string };
}
```